flybywiresim/aircraft
# fix(A380X): Fix throttle calibration for A380X

## Summary of Changes

- Refactored Throttle Calibrationto support A380X
- Fixes throttle calibration and adds throttle events for all throttle axis in the A380X
- Added throttle lever sounds to sound.xml for throttles 3 + 4

- [ ] Remember to also update Localazy!

Discord username (if different from GitHub): cdr_maverick

## Additional Information

This required some refactoring of the existing A32NX calibration which was not intended to be used with 4 axis and throttles. 
Overall this is just a refactor and not very clean in all places. It could use a full refactor so it is easier configurable for the A32NX and A380X.

## Screenshots
To make it clearer which axis nees to be mapped to which throttle the axis header now states for which throttle it is used. 
![image](https://github.com/flybywiresim/aircraft/assets/16833201/973f0742-2554-4fa4-858d-e7f8dff90f7d)
![image](https://github.com/flybywiresim/aircraft/assets/16833201/c4bca278-040b-4530-986b-51a2ee084138)
![image](https://github.com/flybywiresim/aircraft/assets/16833201/8c089097-a067-42be-af55-4a012176ab09)

Example MSFS Mapping for 2 axis mapped to 4 throttles - throttle 1+2 use one axis, 3+4 the other
![image](https://github.com/flybywiresim/aircraft/assets/16833201/b7912f74-3bb3-47f4-a243-5e114e989b7e)

## Testing instructions

General:
- Use Reset To Default and delete the ThrottleConfiguration.ini between tests
- if using less axis then throttles use the following MSFS Controller mapping:
    - A320: 1 axis 2 throttles --> map axis 1 to both throttles (or to the general throttle axis) 
    - A320: 2 axis 2 throttles --> map axis 1 to throttle 1, axis 2 to throttle 2
    - A380: 1 axis 4 throttles --> map axis 1 to all 4 throttles (or to the general throttle axis)
    - A380: 2 axis 4 throttles --> map axis 1 to throttle 1 and 2, map axis 2 to throttle 3 and 4
    - A380: 4 axis 4 throttles --> map axis 1 to throttle 1, 2 to 2, etc.
    
A32NX:
- Calibrate with 1 and 2 axis / reverser on axis or not on axis
- Test Load and Save
- Check the ThrottleConfiguration.ini if it makes sense with your configuration
- 
A380X:
- Calibrate your throttles in the A380X with 1 axis, 2 axis and 4 axis
- Test Load and Save
- Check the ThrottleConfiguration.ini if it makes sense with your configuration

Both: 
- Also check standard throttle events for all 4 axis (e.g. THROTTLE_50, or THROTTLE_SET=-16384..16384, ...) if you have them means (SPAD.neXt, AaO, Mobiflight, etc.)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
